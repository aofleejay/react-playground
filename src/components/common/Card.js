import React, { Fragment } from 'react'
import PropTypes from 'prop-types'

const Card = ({ title, body }) => (
  <Fragment>
    <p id="card-title">{title}</p>
    <p id="card-body">{body}</p>
  </Fragment>
)

Card.defaultProps = {
  title: 'Anonymous',
  body: '',
}

Card.propTypes = {
  title: PropTypes.string,
  body: (props, propName, componentName) => {
    if (props[propName].length > 300) {
      return new Error(
        'Invalid prop `' + propName + '` supplied to' +
        ' `' + componentName + '`. Validation failed.'
      );
    }
  },
}

export default Card
