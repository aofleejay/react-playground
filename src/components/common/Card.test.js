import React from 'react';
import { shallow } from 'enzyme';
import Card from './Card';

describe('Test card ui structure', () => {
  let wrapper;
  let props;

  beforeEach(() => {
    props = {
      title: 'Lorem',
      body: 'Ipsum.',
    };

    wrapper = shallow(<Card {...props} />);
  });

  it('Should render correct card title', () => {
    const cardTitle = wrapper.find('#card-title');

    expect(cardTitle).toHaveLength(1);
    expect(cardTitle.props()).toHaveProperty('children', 'Lorem');
  });

  it('Should contain correct card-body', () => {
    const cardBody = wrapper.find('#card-body');

    expect(cardBody).toHaveLength(1);
    expect(cardBody.props()).toHaveProperty('children', 'Ipsum.');
  });

  describe('Match its snapshot', () => {
    it('When all props exists', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('When no props exists', () => {
      wrapper = shallow(<Card />);
      expect(wrapper).toMatchSnapshot();
    });
  });
});
