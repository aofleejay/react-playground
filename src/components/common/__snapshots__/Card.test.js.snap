// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test card ui structure Match its snapshot When all props exists 1`] = `
<React.Fragment>
  <p
    id="card-title"
  >
    Lorem
  </p>
  <p
    id="card-body"
  >
    Ipsum.
  </p>
</React.Fragment>
`;

exports[`Test card ui structure Match its snapshot When no props exists 1`] = `
<React.Fragment>
  <p
    id="card-title"
  >
    Anonymous
  </p>
  <p
    id="card-body"
  />
</React.Fragment>
`;
